src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -3790,9 +3790,29 @@ void Compiler::fgDebugCheckLoopTable()
         assert(optLoopTable != nullptr);
     }
 
-    // Construct a map of increasing block numbers to allow for block number comparisons when block numbers are not
-    // guaranteed to be in increasing order.
-    unsigned* blockNumMap = fgBuildBlockNumMap();
+    // Build a mapping from existing block list number (bbNum) to the block number it would be after the
+    // blocks are renumbered. This allows making asserts about the relative ordering of blocks using block number
+    // without actually renumbering the blocks, which would affect non-DEBUG code paths. Note that there may be
+    // `blockNumMap[bbNum] == 0` if the `bbNum` block was deleted and blocks haven't been renumbered since
+    // the deletion.
+
+    unsigned bbNumMax = compIsForInlining() ? impInlineInfo->InlinerCompiler->fgBBNumMax : fgBBNumMax;
+
+    // blockNumMap[old block number] => new block number
+    size_t    blockNumBytes = (bbNumMax + 1) * sizeof(unsigned);
+    unsigned* blockNumMap   = (unsigned*)_alloca(blockNumBytes);
```

Comment:
nit: elsewhere in this file, `alloca` is used (instead of `_alloca`).

Reply:
I think `_alloca` is the standard name; I'll just convert everything to use that. Some places use allocacheck.h and its macros, but I don't want to go crazy using that, too.